### PR TITLE
Generic modules for inputs, processing and exporting.

### PR DESCRIPTION
**Pull request by tamir-michaeli**
# About this change - What it does
Adding generic modules for importing influxql dashboard processing metric and exporting the dashboard.
Added file report which will be printed at the end of the run. "replaced_metrics_report.yaml"

### CODE REVIEW

**Comment by tamir-michaeli** [comment]
Remaining changes:
* json schema for the config file and config validation
* add exact match metrics to the result report
* info,debug prints, error handling
* Readme

**Comment by tamir-michaeli** [comment]
Remaining changes and improvements:
* Unit tests
* Json schema for config and config validation
* Improvement: When a templating variable (non host?) is not a valid metric (i.e net - no . or _ in it), look for a valid metric name in the dashboards panels in replace it. (net --> netstat_tcp_close_time)

File: importer/grafana/grafana_importer.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    def fetch_dashboards(self):
+        dashboards = self._build_dashboards_list()
+        return dashboards
+
+    def _extract_dashboard_uids(self, response):
+        response_json = json.loads(response.content)
+        uid_list = []
+        for dashboard in response_json:
+            if dashboard['type'] == 'dash-db':
```
Comment (Doron-Bargo) [inline importer/grafana/grafana_importer.py:29]: Why we choose only this dashboard type

File: processor/find_metrics_names/find_metrics_names_processor.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        metrics = []
+        if response.status_code == 200:
+            metrics = json.loads(response.content).get('data')
+        else:
+            self._logger.error(f"Encountered an error while fetching metrics from db: {json.loads(response.content)}")
+        return metrics
+
+    def order_repalce_strategies(self, strategies):
+        if len(strategies) > 1:
+            if strategies[0] == 'statistic_combination':
```
Comment (Doron-Bargo) [inline processor/find_metrics_names/find_metrics_names_processor.py:166]: Why we change the order?

**Review by Doron-Bargo — APPROVED** [review]
lgtm

File: processor/find_metrics_names/find_metrics_names_processor.py (hunk 2 above)
Comment (tamir-michaeli) [reply processor/find_metrics_names/find_metrics_names_processor.py:166]: Statistic combination requires a lot more resources and time than permutation. Each metric that we find a replacement for it with permutation, saves us a lot of processing time in statistic combination.

**Pull request merged**